apps/browser/src/decorators/session-sync-observable/session-sync.decorator.ts

```diff
@@ -0,0 +1,46 @@
+import { Jsonify } from "type-fest";
+
+import { SessionStorable } from "./session-storable";
+
+class BuildOptions<T> {
+  ctor?: new (args: any[]) => T;
+  initializer?: (key_value_pair: Jsonify<T>) => T;
+  initializeAsArray? = false;
+}
+
+/**
+ * A decorator used to indicate the BehaviorSubject should be synced for this browser session across all contexts.
+ *
+ * >**Note** This decorator does nothing if the enclosing class is not decorated with @browserSession.
+ *
+ * @param buildOptions
+ * Builders for the value, requires either a constructor (ctor) for your BehaviorSubject type or an
+ * initializer function that takes a key value pair representation of the BehaviorSubject data
+ * and returns your BehaviorSubject data type. `initializeAsArray can optionally be used to indicate
```

Comment:
Not sure if `data type` is the right way to describe a `fromJSON` return value. Maybe:

```suggestion
 * and returns your instantiated BehaviorSubject value. `initializeAsArray can optionally be used to indicate
```